### ISSUE TITLE
False positive of `NonAtomicFileOperation` for `FileUtils.rm_r`

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop/rubocop#10813]
From this discussion: https://github.com/rubocop/rubocop/issues/10760#issuecomment-1171707113

## Expected behavior

No offenses or no exceptions for `FileUtils.rm_r`.

## Actual behavior

If I have the code:

```ruby
FileUtils.rm_r gem_name if Dir.exist? gem_name
```

it causes offenses by `NonAtomicFileOperation` cop.

If I remove the condition — it causes the exception:

```ruby
     Failure/Error: FileUtils.rm_r gem_name
     
     Errno::ENOENT:
       No such file or directory @ apply2files - foo_bar
```

Even with `require 'fileutils'`.

## Steps to reproduce the problem

The code is above.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.31.2 (using Parser 3.1.2.0, rubocop-ast 1.19.1, running on ruby 3.1.2 x86_64-linux)
  - rubocop-performance 1.14.2
  - rubocop-rspec 2.12.1
```

**Comment by AlexWayfer** [comment]
@ydah thank you, it's a good change. But… `--force` option is not only for cases where files (or dirs) don't exist. It'll also delete all protected files, probably something else. It's not an equal I beleive.

**Comment by ydah** [comment]
@AlexWayfer Yes, I also certainly don't think they are strictly equivalent. The reason for adding this cop is that these can cause problems that are difficult to reproduce, especially when frequent file operations are performed in parallel, as in test runs using `parallel_rspec`.  However, it might be nice to be able to options for cases allowed where atomic file deletion cannot be safely guaranteed.
